mafiasi/registration/views.py

```diff
@@ -18,38 +18,64 @@
+            form = PrimaryRegisterForm(request.POST)
+        else:
+            form = OtherRegisterForm(request.POST)
         if form.is_valid():
             account = form.cleaned_data['account']
             domain = form.cleaned_data['domain']
             if domain == settings.PRIMARY_DOMAIN:
-                yeargroup = Yeargroup.objects.get_by_account(account)
-                if yeargroup:                     
-                    return _finish_account_request(request, {
-                        'action': 'request_account',
-                        'account': account,
-                        'domain': domain,
-                        'yeargroup_pk': yeargroup.pk
-                    })
-                else:
-                    info_form = AdditionalInfoForm()
-                    info_form.prefill(account, domain)
-                    return TemplateResponse(request,
-                        'registration/require_info.html', {
-                            'account': account,
-                            'info_form': info_form,
-                    })
+                user = get_irz_ldap_entry(uid=account)
+                if user is None:
+                    # Nobody is supposed to know that this account does not exist. Therefore, pretend success.
+                    return TemplateResponse(request, 'registration/request_successful.html')
+                irz_group_slug = get_irz_ldap_group(user['gid'])
+                if irz_group_slug is None:
+                    # this should not happen because all groups used in the LDAP should also exist there
+                    raise AssertionError(f'The yeargroup {user["gid"]} was referenced in the IRZ LDAP '
+                                         'but does not actually exist there.')
+
+                # Our group slugs do not contain the leading 'j'
+                group_slug = irz_group_slug.lstrip('j')
+
+                try:
+                    # Try to find the group with the same gid and slug. Because the gids can be reused
+                    # for new groups, we have to check both.
+                    yeargroup = Yeargroup.objects.get(gid=user['gid'], slug=group_slug)
+                except Yeargroup.DoesNotExist:
+                    if account[0].isdigit():
+                        # This should usually only happen once a year, when the first user
+                        # of the new yeargroup fills out the registration form.
+                        yeargroup = Yeargroup(slug=group_slug, name=group_slug, gid=user['gid'])
+                        yeargroup.save()
+                    else:
+                        # All accounts that start with another letter than a digit are employee
+                        # accounts and should be treated as such. Further differentiation by their
+                        # actual groups (which correspond to the research groups) is not necessary.
+                        yeargroup = Yeargroup.objects.get(slug='employee')
```

Comment:
I changed this to `get_or_create` so that the name can stay `Mitarbeiter` but when a new group is created, it is called `Employees` (e.g. on a new setup). The `slug` has always been `employee`.